### PR TITLE
Add test for InstrumentedEnqueueRequestForObject

### PR DESCRIPTION
**Pull request by jmrodri**
Added a test for the `InstrumentedEnqueueRequestForObject` to increase the code coverage.

Before the test, handler was at 71.2%
```
$ make test                                                          
ok      github.com/operator-framework/operator-lib/handler      1.080s  coverage: 71.2% of statements   
...
```

After adding the test, coverage goes to 100%
```
$ make test                                                  
ok      github.com/operator-framework/operator-lib/handler      1.082s  coverage: 100.0% of statements
...
```

### CODE REVIEW

File: handler/handler_suite_test.go [hunk 1]
```diff
@@ -19,10 +19,9 @@ import (
```
Comment (camilamacedo86) [inline handler/handler_suite_test.go:19]: just a nit .. Add the sufix _test to the package. ( handler_test )

Comment (camilamacedo86) [reply handler/handler_suite_test.go:19]: reason:  the suffix `_test` which has [special meaning for the Go compiler](https://github.com/golang/go/issues/25223).

File: handler/instrumented_enqueue_object_test.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+					Name:      pod.Name,
+				},
+			}))
+
+			gauges, err := metrics.Registry.Gather()
+			Expect(err).Should(BeNil())
+			Expect(len(gauges)).To(Equal(1))
+			for _, l := range gauges[0].Metric[0].Label {
+				if l.Name == ptrString("name") || l.Name == ptrString("namespace") {
+					Expect(l.Value).To(Equal("biz"))
```
Comment (camilamacedo86) [inline handler/instrumented_enqueue_object_test.go:127]: Why we are using `ptrString("name")`?

File: handler/instrumented_enqueue_object_test.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+				}
+			}
```
Comment (camilamacedo86) [inline handler/instrumented_enqueue_object_test.go:70]:
```suggestion
                        name := "name"
                        namespace := "namespace"
			for _, l := range gauges[0].Metric[0].Label {
				if l.Name ==&name || l.Name ==&namespace {
					Expect(l.Value).To(Equal("biz"))
				}
			}
```

Comment (camilamacedo86) [reply handler/instrumented_enqueue_object_test.go:70]: I do not think that we need the `ptrString` in this case.

Comment (camilamacedo86) [reply handler/instrumented_enqueue_object_test.go:70]: Also, name should be == namespace? Should  not it be l.Namespace in the second conditional?

File: handler/instrumented_enqueue_object_test.go [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
```
Comment (joelanford) [inline handler/instrumented_enqueue_object_test.go:129]: Maybe write a helper function for this and also check the metric's GVK labels and the metric value?

**Comment by jmrodri** [comment]
> Just a few nits:
> 
>     * Could we add the suffix `_test` which has [special meaning for the Go compiler](https://github.com/golang/go/issues/25223)? So, it would be `package handler_test` instead of `package handler`
>       `

No, I wanted it in the same package as the code so I could also test the unexported methods. If we put it in `package handler_test` then we can only test the exported method which is great but sometimes it's easier to test the error conditions of the unexported methods directly.

File: handler/instrumented_enqueue_object_test.go [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+	var instance InstrumentedEnqueueRequestForObject
+	var pod *corev1.Pod
+
+	BeforeEach(func() {
+		q = controllertest.Queue{Interface: workqueue.New()}
+		instance = InstrumentedEnqueueRequestForObject{}
+		pod = &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "biz",
+				Name:      "biz",
```
[the remark this answers is not in the document]
Comment (jmrodri) [reply handler/instrumented_enqueue_object_test.go:42]: Excellent idea.

File: handler/instrumented_enqueue_object_test.go [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+			Expect(len(gauges)).To(Equal(1))
+			for _, l := range gauges[0].Metric[0].Label {
+				if l.Name == ptrString("name") || l.Name == ptrString("namespace") {
+					Expect(l.Value).To(Equal("biz"))
+				}
+			}
+		})
+	})
+
+	Describe("Delete", func() {
```
[the remark this answers is not in the document]
Comment (jmrodri) [reply handler/instrumented_enqueue_object_test.go:82]: @joelanford I added contexts, excellent tip.

**Pull request merged**